pkg/endpointmanager/policymap_pressure.go

```diff
@@ -0,0 +1,65 @@
+package endpointmanager
+
+import (
+	"time"
+
+	"github.com/cilium/cilium/pkg/endpoint"
+	"github.com/cilium/cilium/pkg/maps/policymap"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/stream"
+	"github.com/cilium/cilium/pkg/trigger"
+)
+
+func (mgr *endpointManager) initPolicyMapPressureMetric() {
+	if !metrics.BPFMapPressure {
+		return
+	}
+	mgr.policyMapPressureGauge = metrics.NewBPFMapPressureGauge(policymap.MapName, 0.0)
+
+	go mgr.readPolicyMapPressureEvents()
+
+	var err error
+	src, emit, _ := stream.Multicast[endpoint.MapPressureEvent]() // ignoring complete() here as there's no single emitter that can complete it
+	mgr.policyMapPressureEventCh = stream.ToChannel(mgr.ctx, src)
+	mgr.policyMapPressureEmitFunc = emit
+	mgr.policyMapPressureTrigger, err = trigger.NewTrigger(trigger.Parameters{
+		MinInterval:  10 * time.Second,
+		TriggerFunc:  func([]string) { mgr.updatePolicyMapPressureMetric() },
+		ShutdownFunc: func() {},
+		Name:         "endpointmanager-policymap-max-size-metrics",
+	})
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (mgr *endpointManager) readPolicyMapPressureEvents() {
+	for {
+		select {
+		case ev := <-mgr.policyMapPressureEventCh:
+			val := ev.Value
+			log.Infof("EndpointManager policymap received event: %s", val)
+			cur := mgr.currentPolicyMapPressure.Load()
+			// TODO: Do we need to define a threshold for the comparison?
+			if cur == nil || val > *cur {
+				mgr.currentPolicyMapPressure.Store(&val)
+				mgr.policyMapPressureTrigger.Trigger()
+			}
+		case <-mgr.ctx.Done():
+			log.Infof("EndpointManager policymap event loop terminated")
```

Comment:
This won't drain the channel and will block the emitter.

Reply:
Removed channel and usage of `pkg/stream`